modules/concurrency/include/hephaestus/concurrency/io_ring.h

```diff
@@ -0,0 +1,60 @@
+//=================================================================================================
+// Copyright (C) 2023-2025 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <liburing.h>
+#include <stdexec/stop_token.hpp>
+
+#include "hephaestus/concurrency/io_ring_operation_pointer.h"
+
+namespace heph::concurrency {
+
+struct IoRingConfig {
+  static constexpr unsigned DEFAULT_NENTRIES = 1024;
+  unsigned nentries{ DEFAULT_NENTRIES };
+  unsigned flags{ IORING_SETUP_DEFER_TASKRUN | IORING_SETUP_SINGLE_ISSUER };
```

Comment:
nit. I would use `uint32_t` to be more explicit
```suggestion
  uint32_t nentries{ DEFAULT_ENTRIES_COUNT };
  uint32_t flags{ IORING_SETUP_DEFER_TASKRUN | IORING_SETUP_SINGLE_ISSUER };
```